packages/boilerplate-generator/boilerplate_web_browser_template.js

```diff
@@ -0,0 +1,70 @@
+// Template function for rendering the boilerplate html for browsers
+// Replicates the template defined in boilerplate_web.browser.html
+// Arguments: root : { htmlAttributes, css : [{ url }], bundledJsCssUrlRewriteHook : Function, head, dynamicHead, body, dynamicBody, inlineScriptsAllowed, additionalStaticJs, meteorRuntimeConfig }
+
+export default function (manifest) {
+  const root = manifest;
+  // XXX do we need to do some validation on the properties of root?
+  return [].concat(
+    [
+      '<html' +_.map(root.htmlAttributes, (value, key) =>
+        _.template(' <%= attrName %>="<%- attrValue %>"')({
+          attrName: key,
+          attrValue: value
+        })
+      ).join('') + '>',
+      '<head>'
+    ],
+
+    _.map(root.css, ({url}) =>
+      _.template('  <link rel="stylesheet" type="text/css" class="__meteor-css__" href="<%- href %>">')({
+        href: root.bundledJsCssUrlRewriteHook(url)
+      })
+    ),
+
+    [
+      root.head,
+      root.dynamicHead,
+      '</head>',
+      '<body>',
+      root.body,
+      root.dynamicBody,
+      '',
+      _.template(root.inlineScriptsAllowed
+        ? '  <script type="text/javascript"><%= contents %></script>'
+        : '  <script type="text/javascript" src="<%- src %>"></script>'
+      )({
+        contents: _.template('__meteor_runtime_config__ = JSON.parse(decodeURIComponent(<%= conf %>))')({
+          conf: root.meteorRuntimeConfig
+        }),
+        src: root.rootUrlPathPrefix + '/meteor_runtime_config.js'
+      }),
+      ''
+    ],
+
+    _.map(root.js, ({url}) =>
+      _.template('  <script type="text/javascript" src="<%- src %>"></script>')({
+        src: root.bundledJsCssUrlRewriteHook(url)
+      })
+    ),
+
+    _.map(root.additionalStaticJs, ({contents, pathname}) => (
+      _.template(root.inlineScriptsAllowed
+        ? '  <script><%= contents %></script>'
```

Comment:
Discussed this with @benjamn, we decided that it's probably safest to load all `additionalStaticJs` from the server. The reason is that embedding arbitrary javascript inline is tricky to do correctly, and can cause strange bugs.

The tradeoff we're making here is that loading from the server may be more costly than inlining the script. The only internal package that is loaded through `additionalStaticJs` is https://github.com/meteor/meteor/tree/master/packages/reload-safetybelt, so hopefully this change won't be too disruptive?